### ISSUE TITLE
Forbid too deep subscript and attribute usages

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#737]
# Rule request

## Thesis

Code like:

```python
print(my_matrix[0][0][0][0][0][0][0][1])
```

And code like:

```python
self.attr.inner.wrapper.field.value

self.attr.inner.wrapper.method.call()
```

When counting access level we count:
1. `ast.Subscript`
2. `ast.Attribute`

Things we do not count:
1. `ast.Index` complexity
2. `ast.Call` (since there are a lot of `call`-based APIs like `django` ORM: `.filter().exclude().annotate().values().first()` which we do not want to break)

That's how it works:

```python
obj.attr.call().other[0].field.type.boom
# obj has access level of one: `.attr`
# `call()` has access level of 5: `.other`, `[0]`, `.field`, `.type`, `.boom`
```

Should be forbidden. We need to make this complexity rule configurable.
Default value can be `4`, chains longer than `4` should raise a violation.

## Reasoning

It indicates a bad design and overcomplicated data without proper API.

**Comment by artemiy312** [comment]
Hi @sobolevn! May I take the issue?

**Comment by artemiy312** [comment]
@sobolevn Should we forbid non-consecutive deep subscript usages? This expression won't break the rule, but it is still overcomplicated data without API: `my_matrix[0][0][0].foo[0][0][0].bar[0][1]`

**Comment by sobolevn** [comment]
@artemiy312 good catch. Maybe we can join two rules together? This one and #738 
Any too long access should be forbidden.

**Comment by artemiy312** [comment]
@sobolevn Sure, let's join these rules

**Comment by sobolevn** [comment]
@artemiy312 have a look at this module: https://github.com/wemake-services/wemake-python-styleguide/tree/master/wemake_python_styleguide/logic
